src/Serilog/Formatting/Display/MessageTemplateTextFormatter.cs

```diff
@@ -58,42 +61,70 @@ public void Format(LogEvent logEvent, TextWriter output)
             if (logEvent == null) throw new ArgumentNullException(nameof(logEvent));
             if (output == null) throw new ArgumentNullException(nameof(output));
 
-            // This could be lazier: the output properties include
-            // everything from the log event, but often we won't need any more than
-            // just the standard timestamp/message etc.
-            var outputProperties = OutputProperties.GetOutputProperties(logEvent, _outputTemplate);
```

Comment:
The changes in this method are the meat of the "less allocations" change - `GetOutputProperties()` above would previously allocate objects for `{Level}`, `{Message}`, `{Exception}` etc. Instead now we just iterate and examine property names, and write the output inline.